### PR TITLE
feat: support CAPTCHA action via Casdoor

### LINKED ISSUE #37: [feature] support CAPTCHA action provided by Casdoor
Casdoor supports CAPTCHA in its login page: https://casdoor.org/docs/provider/captcha/overview

In: https://door.caswaf.com/sites/admin/casdoor_demo, we can require the request to the site to pass the CAPTCHA like the existing passing auth.

image

Maybe Casdoor needs to be updated too to provide a clean CAPTCHA widget.

### PR DESCRIPTION
**Pull request by love98ooo**
Casdoor provides a clean CAPTCHA widget after https://github.com/casdoor/casdoor/pull/3144. Make sure Casdoor version >= 1.681.0 to use CAPTCHA action.

Fix: https://github.com/casbin/caswaf/issues/37

### CODE REVIEW

File: service/proxy.go [hunk 1]
```diff
@@ -214,6 +214,16 @@ func handleRequest(w http.ResponseWriter, r *http.Request) {
...
 			responseError(w, "Dropped by CasWAF: %s", reason)
 			w.WriteHeader(http.StatusBadRequest)
+		case "Captcha":
+			ok := isVerifiedSession(r)
+			if ok {
+				w.WriteHeader(http.StatusOK)
+				nextHandle(w, r)
+				return
+			}
+			w.Header().Set("Set-Cookie", "casdoor_captcha_token=; Path=/; Max-Age=-1")
```
Comment (nomeguy) [inline service/proxy.go:224]: Use `c.Ctx.SetCookie`

File: service/captcha.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
+package service
```
Comment (nomeguy) [inline service/captcha.go:15]: Apache header

File: go.mod [hunk 3]
```diff
@@ -13,6 +13,7 @@ require (
 	github.com/go-git/go-git/v5 v5.9.0
 	github.com/go-mysql-org/go-mysql v1.7.0
 	github.com/go-sql-driver/mysql v1.6.0
+	github.com/google/uuid v1.3.0
```
Comment (nomeguy) [inline go.mod]: go.mod not updated

**Review by nomeguy — CHANGES_REQUESTED** [review]

File: service/proxy.go (hunk 1 above)
Comment (love98ooo) [reply service/proxy.go:224]: Context is not used in the function.

**Review by nomeguy — APPROVED** [review]

**Pull request merged**